<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>GSVA analysis human DLPFC data</title>

<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       </style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/default.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<script src="site_libs/kePrint-0.0.1/kePrint.js"></script>
<link href="site_libs/lightable-0.0.1/lightable.css" rel="stylesheet" />

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
  p.abstract{
    text-align: center;
    font-weight: bold;
  }
  div.abstract{
    margin: auto;
    width: 90%;
  }
</style>



<link rel="stylesheet" href="/home/peremoles/R/x86_64-pc-linux-gnu-library/4.4/BiocStyle/resources/html/bioconductor.css" type="text/css" />



<style type = "text/css">
.main-container {
  max-width: 828px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {

}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 246px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



<script>
function toggle_visibility(id1) {
  var e = document.getElementById(id1);
  e.style.display = ((e.style.display!="none") ? "none" : "block");
}
</script>

</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">GSVA analysis of spatial transcriptomics data</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="03_qa_pr_hdlpfc.html">QA &amp; processing for human DLPFC data</a>
</li>
<li>
  <a href="04_gsva_hdlpfc.html">Spatial and GSVA analysis for human DLPFC data</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">GSVA analysis human DLPFC data</h1>
<p class="author-name">Pere Moles<span class="affil-mark">1*</span></p>
<p class="author-affiliation"><span class="affil-mark">1</span>MSc Data Science student, Universitat Oberta de Catalunya</p>
<p class="author-email"><span class="affil-mark">*</span><a href="mailto:pere.moles@uoc.edu">pere.moles@uoc.edu</a></p>
<h4 class="date">de juny 4, 2024</h4>
<h4 class="abstract">Abstract</h4>
<p>Here we perform a GSVA analysis of the 10x Visium human DLPFC data.</p>

</div>


<div id="import-processed-data" class="section level1" number="1">
<h1><span class="header-section-number">1</span> Import processed data</h1>
<p>Start by importing processed data and gene sets.</p>
<pre class="r"><code>library(SpatialExperiment)

spe.filt &lt;- readRDS(file.path(&quot;_processed_data&quot;, &quot;spe.filt.human_dlpfc.rds&quot;))
dim(spe.filt)
[1] 11364  3591</code></pre>
<pre class="r"><code>
cell_types &lt;- readRDS(file.path(&quot;_processed_data&quot;, &quot;cell_types.human_dlpfc.rds&quot;))</code></pre>
</div>
<div id="clustering" class="section level1" number="2">
<h1><span class="header-section-number">2</span> Clustering</h1>
<div id="pca-based-clustering" class="section level2" number="2.1">
<h2><span class="header-section-number">2.1</span> PCA Based Clustering</h2>
<p>Select a subset of top highly-variable genes (HVGs).</p>
<pre class="r"><code>library(scran)
S&#39;està carregant el paquet requerit: scuttle</code></pre>
<pre class="r"><code>
dec &lt;- modelGeneVar(spe.filt, assay.type=&quot;logcounts&quot;)</code></pre>
<p>Figure <a href="#fig:hdlpfcmeanvar">1</a> shows the mean-variance relationship.</p>
<pre class="r"><code>fit &lt;- metadata(dec)
plot(fit$mean, fit$var, xlab=&quot;mean of log-expression&quot;,
     ylab=&quot;variance of log-expression&quot;)
curve(fit$trend(x), col=&quot;dodgerblue&quot;, add=TRUE, lwd=2)</code></pre>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:hdlpfcmeanvar"></span>
<img src="04_gsva_hdlpfc_files/figure-html/hdlpfcmeanvar-1.png" alt="Mean-variance relationship." width="500px"  class="widefigure" />
<p class="caption">
Figure 1: <span class="caption-title">Mean-variance relationship</span><br>
</p>
</div>
<p>Select top 10% HVGs.</p>
<pre class="r"><code>top_hvgs &lt;- getTopHVGs(dec, prop=0.1)
length(top_hvgs)
[1] 831</code></pre>
<p>Calculate PCA.</p>
<pre class="r"><code>library(scater)
S&#39;està carregant el paquet requerit: ggplot2</code></pre>
<pre class="r"><code>
set.seed(123)
spe.filt &lt;- runPCA(spe.filt, subset_row=top_hvgs)

reducedDimNames(spe.filt)
[1] &quot;PCA&quot;               &quot;TSNE_perplexity50&quot; &quot;TSNE_perplexity5&quot; 
[4] &quot;TSNE_perplexity20&quot; &quot;TSNE_perplexity80&quot; &quot;UMAP_neighbors15&quot; </code></pre>
<pre class="r"><code>dim(reducedDim(spe.filt, &quot;PCA&quot;))
[1] 3591   50</code></pre>
<p>Here we perform a non-spatial graph-based clustering of the spots,
targeting 7 clusters, which is the number of anatomically annotated
groups of spots (six cortical layers and white matter) stored in the
column <code>spatialLIBD</code>.</p>
<pre class="r"><code>table(colData(spe.filt)$spatialLIBD)

 L1  L2  L3  L4  L5  L6  WM 
262 251 987 218 671 689 513 </code></pre>
<pre class="r"><code>k &lt;- 19
g &lt;- buildSNNGraph(spe.filt, k=k, use.dimred=&quot;PCA&quot;)
set.seed(123)
g_walk &lt;- igraph::cluster_walktrap(g)
colLabels(spe.filt) &lt;- factor(g_walk$membership)
table(colLabels(spe.filt))

   1    2    3    4    5    6    7 
 293  664 1177  304  830  221  102 </code></pre>
<p>Figure <a href="#fig:hdlpfcclustering">2</a> shows the clustering of the spots.</p>
<pre class="r"><code>library(RColorBrewer)
library(patchwork)
library(ggspavis)

set.seed(123)
## image seems to be reversed and for this reason we set x_coord=&quot;pxl_col_in_fullres&quot;
## and y_coord=&quot;pxl_row_in_fullres&quot;
pltgt &lt;- plotVisium(spe.filt, x_coord=&quot;pxl_col_in_fullres&quot;,
                    y_coord=&quot;pxl_row_in_fullres&quot;, point_size=0.75,
                    annotate=&quot;spatialLIBD&quot;,
                    pal=brewer.pal(nlevels(colLabels(spe.filt)), &quot;Set1&quot;))
pltcl &lt;- plotVisium(spe.filt, x_coord=&quot;pxl_col_in_fullres&quot;,
                    y_coord=&quot;pxl_row_in_fullres&quot;, point_size=0.75, annotate=&quot;label&quot;,
                    pal=brewer.pal(nlevels(colLabels(spe.filt)), &quot;Set1&quot;))
(pltgt | pltcl)</code></pre>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:hdlpfcclustering"></span>
<img src="04_gsva_hdlpfc_files/figure-html/hdlpfcclustering-1.png" alt="Clustering of the spots." width="800px"  class="widefigure" />
<p class="caption">
Figure 2: <span class="caption-title">Clustering of the spots</span><br>
</p>
</div>
</div>
<div id="spatialpca-based-clustering" class="section level2" number="2.2">
<h2><span class="header-section-number">2.2</span> SpatialPCA based clustering</h2>
<pre class="r"><code>library(SpatialPCA)
LIBD &lt;- CreateSpatialPCAObject(counts=as.matrix(assays(spe.filt)$counts), location=spatialCoords(spe.filt), project = &quot;SpatialPCA&quot;,gene.type=&quot;spatial&quot;,sparkversion=&quot;sparkx&quot;,numCores_spark=5, customGenelist=NULL,min.loctions = 20, min.features=20)
## Use SCTransform function in Seurat to normalize data. 
## Use sparkx function in SPARK to select spatially variable genes. 
## ===== SPARK-X INPUT INFORMATION ==== 
## number of total samples: 3591 
## number of total genes: 11364 
## Running with 5 cores 
## Testing With Projection Kernel
## Testing With Gaussian Kernel 1
## Testing With Gaussian Kernel 2
## Testing With Gaussian Kernel 3
## Testing With Gaussian Kernel 4
## Testing With Gaussian Kernel 5
## Testing With Cosine Kernel 1
## Testing With Cosine Kernel 2
## Testing With Cosine Kernel 3
## Testing With Cosine Kernel 4
## Testing With Cosine Kernel 5
## Identified  10097  spatial genes through SPARK-X function. 
## Using top  3000  significant spatially variable genes. </code></pre>
<pre class="r"><code>
start_time &lt;- Sys.time()
LIBD = SpatialPCA_buildKernel(LIBD, kerneltype=&quot;gaussian&quot;, bandwidthtype=&quot;SJ&quot;,bandwidth.set.by.user=NULL)
## Selected kernel type is:  gaussian  
## The bandwidth is:  0.0458021928511147  
## Calculating kernel matrix
## Finished calculating kernel matrix.</code></pre>
<pre class="r"><code>LIBD = SpatialPCA_EstimateLoading(LIBD,fast=FALSE,SpatialPCnum=20)
[1] &quot;Eigen decomposition on kernel matrix!&quot;
[1] &quot;Using all eigenvectors and eigenvalues in the Kernel matrix!&quot;</code></pre>
<pre class="r"><code>LIBD = SpatialPCA_SpatialPCs(LIBD, fast=FALSE)
end_time &lt;- Sys.time()
T = end_time - start_time
T
Time difference of 2.098107 mins</code></pre>
<pre class="r"><code>clusterlabel= walktrap_clustering(clusternum=7,latent_dat=LIBD@SpatialPCs,knearest=70 ) 
# for other Visium or ST data, the user can also set k nearest number as round(sqrt(dim(SpatialPCAobject@SpatialPCs)[2])) by default.
clusterlabel_refine = refine_cluster_10x(clusterlabels=clusterlabel,location=LIBD@location,shape=&quot;hexagon&quot;)</code></pre>
<pre class="r"><code>cbp=c(&quot;#9C9EDE&quot; ,&quot;#5CB85C&quot; ,&quot;#E377C2&quot;, &quot;#4DBBD5&quot; ,&quot;#FED439&quot; ,&quot;#FF9896&quot;, &quot;#FFDC91&quot;)
spclus &lt;- plot_cluster(location=cbind(-spatialCoords(spe.filt)[,2],spatialCoords(spe.filt)[,1]),clusterlabel=clusterlabel_refine,pointsize=1.5 ,title_in=paste0(&quot;SpatialPCA clustering of Human DLPFC data&quot;),color_in=cbp)
spclus</code></pre>
<p><img src="04_gsva_hdlpfc_files/figure-html/unnamed-chunk-9-1.png" width="100%"  class="widefigure" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="gsva-enrichment-score-estimation" class="section level1" number="3">
<h1><span class="header-section-number">3</span> GSVA enrichment score estimation</h1>
<p>Run GSVA with default parameters.</p>
<pre class="r"><code>library(GSVA)
library(BiocParallel)

gsvaPar &lt;- gsvaParam(spe.filt, cell_types, assay=&quot;logcounts&quot;,
                     minSize=5, kcdf = &#39;none&#39;)
spe.filt.es &lt;- gsva(gsvaPar, verbose=FALSE, BPPARAM=MulticoreParam(workers=3))
dim(spe.filt.es)
[1]   50 3591</code></pre>
<pre class="r"><code>spe.filt.es
class: SpatialExperiment 
dim: 50 3591 
metadata(0):
assays(1): es
rownames(50): Astrocyte CD4..T.cell ... SCZ.PGC.GWAS SCZ.SNV
rowData names(1): gs
colnames(3591): AAACAAGTATCTCCCA-1 AAACAATCTACTAGCA-1 ...
  TTGTTTGTATTACACG-1 TTGTTTGTGTAAATTC-1
colData names(80): sample_id Cluster ... sizeFactor label
reducedDimNames(0):
mainExpName: NULL
altExpNames(0):
spatialCoords names(2) : pxl_col_in_fullres pxl_row_in_fullres
imgData names(4): sample_id image_id data scaleFactor</code></pre>
</div>
<div id="spatially-variable-gene-sets" class="section level1" number="4">
<h1><span class="header-section-number">4</span> Spatially Variable Gene Sets</h1>
<pre class="r"><code>library(nnSVG)

system.time(spe.nnsvg &lt;- nnSVG(spe.filt[top_hvgs,], assay_name = &quot;logcounts&quot;, BPPARAM=MulticoreParam(workers=5)))
       usuari       sistema transcorregut 
     2931.100      4073.542      1001.523 </code></pre>
<pre class="r"><code>system.time(spe.nnsvg.es &lt;-nnSVG(spe.filt.es, assay_name = &quot;es&quot;, BPPARAM=MulticoreParam(workers=4)))
       usuari       sistema transcorregut 
      203.465       294.224        76.814 </code></pre>
<pre class="r"><code># number of significant SV Genes and Gene Sets
table(rowData(spe.nnsvg)$padj &lt;= 0.05)

FALSE  TRUE 
   92   739 </code></pre>
<pre class="r"><code>table(rowData(spe.nnsvg.es)$padj &lt;= 0.05)

FALSE  TRUE 
    3    47 </code></pre>
<pre class="r"><code>
rowData(spe.nnsvg)[order(rowData(spe.nnsvg)$rank)[1:5], ]
DataFrame with 5 rows and 23 columns
                        source     type         gene_id gene_version
                      &lt;factor&gt; &lt;factor&gt;     &lt;character&gt;  &lt;character&gt;
ENSG00000197971 ensembl_havana     gene ENSG00000197971           14
ENSG00000123560 ensembl_havana     gene ENSG00000123560           13
ENSG00000198804 insdc              gene ENSG00000198804            2
ENSG00000110484 ensembl_havana     gene ENSG00000110484            6
ENSG00000131095 ensembl_havana     gene ENSG00000131095           12
                  gene_name    gene_source   gene_biotype
                &lt;character&gt;    &lt;character&gt;    &lt;character&gt;
ENSG00000197971         MBP ensembl_havana protein_coding
ENSG00000123560        PLP1 ensembl_havana protein_coding
ENSG00000198804      MT-CO1          insdc protein_coding
ENSG00000110484     SCGB2A2 ensembl_havana protein_coding
ENSG00000131095        GFAP ensembl_havana protein_coding
                           gene_search is_top_hvg   sigma.sq    tau.sq
                           &lt;character&gt;  &lt;logical&gt;  &lt;numeric&gt; &lt;numeric&gt;
ENSG00000197971   MBP; ENSG00000197971       TRUE  1.4700626  0.387177
ENSG00000123560  PLP1; ENSG00000123560       TRUE  1.0063715  0.623055
ENSG00000198804 MT-CO1; ENSG00000198..       TRUE  0.0937191  0.053727
ENSG00000110484 SCGB2A2; ENSG0000011..       TRUE 18.1109288  0.365768
ENSG00000131095  GFAP; ENSG00000131095       TRUE  0.7006729  0.519666
                      phi    loglik   runtime      mean       var     spcov
                &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt;
ENSG00000197971   1.35424 -3671.416     3.251   3.53333  2.482468 0.3431501
ENSG00000123560  10.48222 -4741.616     3.712   2.58666  2.896605 0.3878288
ENSG00000198804  11.17711  -385.219     1.329   7.02010  0.226935 0.0436085
ENSG00000110484   1.53764 -4588.685     3.406   1.86094  2.338538 2.2868556
ENSG00000131095  16.58370 -4500.583     1.730   1.85011  2.060159 0.4524400
                  prop_sv loglik_lm   LR_stat      rank      pval      padj
                &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt;
ENSG00000197971  0.791531  -6727.47   6112.11         1         0         0
ENSG00000123560  0.617623  -7004.49   4525.75         2         0         0
ENSG00000198804  0.635616  -2432.02   4093.59         3         0         0
ENSG00000110484  0.980204  -6620.23   4063.09         4         0         0
ENSG00000131095  0.574162  -6392.67   3784.17         5         0         0</code></pre>
<pre class="r"><code>rowData(spe.nnsvg.es)[order(rowData(spe.nnsvg.es)$rank)[1:10], ]
DataFrame with 10 rows and 15 columns
                                                                                 gs
                                                                    &lt;CharacterList&gt;
Mature.oligodendrocyte          ENSG00000204655,ENSG00000123560,ENSG00000173786,...
Oligodendrocyte                 ENSG00000175206,ENSG00000175130,ENSG00000116133,...
Microglial.cell                 ENSG00000078808,ENSG00000116251,ENSG00000142583,...
Mid.radical.glial.cell          ENSG00000197921,ENSG00000173406,ENSG00000132688,...
Astrocyte                       ENSG00000117318,ENSG00000177606,ENSG00000162599,...
Glial.cell                      ENSG00000160678,ENSG00000079215,ENSG00000204655,...
Lake.et.al.Science.Ex1          ENSG00000115756,ENSG00000135919,ENSG00000113100,...
Oligodendrocyte.progenitor.cell ENSG00000132692,ENSG00000144230,ENSG00000134853,...
Neuron                          ENSG00000020129,ENSG00000162409,ENSG00000117569,...
Gene_SFARI_all                  ENSG00000183044,ENSG00000064687,ENSG00000087085,...
                                   sigma.sq      tau.sq         phi    loglik
                                  &lt;numeric&gt;   &lt;numeric&gt;   &lt;numeric&gt; &lt;numeric&gt;
Mature.oligodendrocyte          0.051760089 0.020496709  8.25637981   1319.18
Oligodendrocyte                 0.018177771 0.001817777  1.00000000   5856.86
Microglial.cell                 0.002211274 0.001377564 32.18139858   5767.56
Mid.radical.glial.cell          0.004124360 0.004480564 27.96337498   4001.28
Astrocyte                       0.002643844 0.002617861 25.24216947   4964.10
Glial.cell                      0.015332265 0.025606447  7.21814535   1240.54
Lake.et.al.Science.Ex1          0.006647546 0.012231093 11.53036223   2530.57
Oligodendrocyte.progenitor.cell 0.008961859 0.020472383  0.00202358   1745.73
Neuron                          0.001132983 0.001731306 38.76953385   5742.74
Gene_SFARI_all                  0.000229339 0.000541473 14.68968087   8137.07
                                  runtime      mean       var     spcov
                                &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt;
Mature.oligodendrocyte              5.491        NA        NA        NA
Oligodendrocyte                     1.154        NA        NA        NA
Microglial.cell                     3.907        NA        NA        NA
Mid.radical.glial.cell              5.590        NA        NA        NA
Astrocyte                           4.472        NA        NA        NA
Glial.cell                          4.963        NA        NA        NA
Lake.et.al.Science.Ex1              1.616        NA        NA        NA
Oligodendrocyte.progenitor.cell     5.955        NA        NA        NA
Neuron                              6.147        NA        NA        NA
Gene_SFARI_all                      2.962        NA        NA        NA
                                  prop_sv loglik_lm   LR_stat      rank
                                &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt;
Mature.oligodendrocyte           0.716335 -2111.052   6860.47         1
Oligodendrocyte                  0.909091  2962.735   5788.25         2
Microglial.cell                  0.616153  4278.861   2977.40         3
Mid.radical.glial.cell           0.479302  2866.222   2270.12         4
Astrocyte                        0.502469  3878.357   2171.48         5
Glial.cell                       0.374518   343.234   1794.61         6
Lake.et.al.Science.Ex1           0.352120  1802.148   1456.85         7
Oligodendrocyte.progenitor.cell  0.304471  1070.772   1349.91         8
Neuron                           0.395555  5126.095   1233.30         9
Gene_SFARI_all                   0.297529  7565.368   1143.41        10
                                     pval      padj
                                &lt;numeric&gt; &lt;numeric&gt;
Mature.oligodendrocyte                  0         0
Oligodendrocyte                         0         0
Microglial.cell                         0         0
Mid.radical.glial.cell                  0         0
Astrocyte                               0         0
Glial.cell                              0         0
Lake.et.al.Science.Ex1                  0         0
Oligodendrocyte.progenitor.cell         0         0
Neuron                                  0         0
Gene_SFARI_all                          0         0</code></pre>
<p>Let’s check how the genes in the most spatially variable gene sets behave.</p>
<pre class="r"><code># Most spatially variable gene set

for(j in 1:4){
  gs &lt;- as.character(unlist(rowData(spe.nnsvg.es)[order(rowData(spe.nnsvg.es)$rank)[j], &quot;gs&quot;]))
  svgset &lt;- rownames(rowData(spe.nnsvg.es)[order(rowData(spe.nnsvg.es)$rank)[j], ])

  print(paste(&quot;Length of the gene set&quot;,svgset,length(gs),&quot;genes&quot;))
  svgs &lt;- rowData(spe.nnsvg)[gs,]
  svgs &lt;- svgs[order(svgs$rank),][1:5,]
  i&lt;-1
  plts &lt;- list()
  for (g in rownames(svgs)) {
    plts[[svgset]] &lt;- plotVisium(spe.filt.es, x_coord=&quot;pxl_col_in_fullres&quot;,
                             y_coord=&quot;pxl_row_in_fullres&quot;, point_size=0.5,
                             assay=&quot;es&quot;, annotate=svgset, facets=NULL) +
                        ggtitle(svgset) + theme(plot.title=element_text(hjust=0.5)) +
                        theme(legend.title=element_blank()) + labs(tag=letters[i])
    plts[[g]] &lt;- plotVisium(spe.filt, x_coord=&quot;pxl_col_in_fullres&quot;,
                             y_coord=&quot;pxl_row_in_fullres&quot;, point_size=0.5,
                             assay=&quot;logcounts&quot;, annotate=g, facets=NULL) +
                        ggtitle(g) + theme(plot.title=element_text(hjust=0.5)) +
                        theme(legend.title=element_blank()) + labs(tag=letters[i])
    i &lt;- i + 1
  }
  show(wrap_plots(plts, nrow=ceiling(length(plts)/2)))
}
[1] &quot;Length of the gene set Mature.oligodendrocyte 5 genes&quot;</code></pre>
<p><img src="04_gsva_hdlpfc_files/figure-html/unnamed-chunk-15-1.png" width="100%"  class="widefigure" style="display: block; margin: auto;" /></p>
<pre><code>[1] &quot;Length of the gene set Oligodendrocyte 165 genes&quot;</code></pre>
<p><img src="04_gsva_hdlpfc_files/figure-html/unnamed-chunk-15-2.png" width="100%"  class="widefigure" style="display: block; margin: auto;" /></p>
<pre><code>[1] &quot;Length of the gene set Microglial.cell 250 genes&quot;</code></pre>
<p><img src="04_gsva_hdlpfc_files/figure-html/unnamed-chunk-15-3.png" width="100%"  class="widefigure" style="display: block; margin: auto;" /></p>
<pre><code>[1] &quot;Length of the gene set Mid.radical.glial.cell 20 genes&quot;</code></pre>
<p><img src="04_gsva_hdlpfc_files/figure-html/unnamed-chunk-15-4.png" width="100%"  class="widefigure" style="display: block; margin: auto;" />
# Moran’s I estimation</p>
<p>Here we calculate Moran’s I for each gene set in the GSVA
enrichment scores (ES) calculated with default parameters.</p>
<pre class="r"><code>library(SpatialFeatureExperiment)
library(Voyager)

sfe &lt;- toSpatialFeatureExperiment(spe.filt.es)
vg &lt;- findVisiumGraph(spe.filt.es,
                      sample_id=&quot;151673&quot;,
                      zero.policy=TRUE)
colGraph(sfe, &quot;visium&quot;, sample_id=&quot;151673&quot;) &lt;- vg

moransi &lt;- rowData(runUnivariate(sfe, type=&quot;moran.mc&quot;, nsim = 1000, colGraphName=&quot;visium&quot;,
                                 exprs_values=&quot;es&quot;, zero.policy=TRUE))
moransi &lt;- moransi[, c(2,4)]
colnames(moransi) &lt;- c(&quot;statistic&quot;, &quot;p.value&quot;)</code></pre>
<div id="sfari-gene-sets-analysis" class="section level2" number="4.1">
<h2><span class="header-section-number">4.1</span> SFARI gene sets analysis</h2>
<p>Figure <a href="#fig:sfarigenesets">3</a> shows GSVA scores for SFARI gene sets.</p>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:sfarigenesets"></span>
<img src="04_gsva_hdlpfc_files/figure-html/sfarigenesets-1.png" alt="GSVA scores for SFARI gene sets" width="800px"  class="widefigure" />
<p class="caption">
Figure 3: <span class="caption-title">GSVA scores for SFARI gene sets</span><br>
</p>
</div>
<p>SFARI gene sets spatial autocorrelation</p>
<pre class="r"><code>moransi[names(cell_types[c(96:98)]),]
DataFrame with 3 rows and 2 columns
                     statistic     p.value
                     &lt;numeric&gt;   &lt;numeric&gt;
Gene_SFARI_all        0.350488 0.000999001
Gene_SFARI_high       0.286730 0.000999001
Gene_SFARI_syndromic  0.163606 0.000999001</code></pre>
</div>
<div id="birnbaum-gene-sets-analysis" class="section level2" number="4.2">
<h2><span class="header-section-number">4.2</span> Birnbaum gene sets analysis</h2>
<p>Figure <a href="#fig:birnbaumgenesets">4</a> shows GSVA scores for Birnbaum et al gene sets.</p>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:birnbaumgenesets"></span>
<img src="04_gsva_hdlpfc_files/figure-html/birnbaumgenesets-1.png" alt="GSVA scores for Birnbaum et al gene sets." width="800px"  class="widefigure" />
<p class="caption">
Figure 4: <span class="caption-title">GSVA scores for Birnbaum et al gene sets</span><br>
</p>
</div>
<pre class="r"><code>moransi[names(cell_types[c(99:108)]),]
DataFrame with 10 rows and 2 columns
                  statistic     p.value
                  &lt;numeric&gt;   &lt;numeric&gt;
ASD.CNV           0.1085017 0.000999001
ASD.DATABASE      0.1803301 0.000999001
BPAD.GWAS         0.0445744 0.000999001
ID                0.1154954 0.000999001
NDD               0.0116347 0.106893107
Neurodegenerative 0.0873586 0.000999001
SCZ.CNV           0.1059468 0.000999001
SCZ.Meta.analysis 0.1044286 0.000999001
SCZ.PGC.GWAS      0.0758406 0.000999001
SCZ.SNV           0.0678499 0.000999001</code></pre>
</div>
<div id="selection-of-autocorrelated-gene-sets" class="section level2" number="4.3">
<h2><span class="header-section-number">4.3</span> Selection of autocorrelated gene sets</h2>
<p>Table <a href="#tab:gsbymoransi">1</a> below shows the subset of gene sets with
significant nonzero Moran’s I, in decreasing order, calculated from GSVA
ES with default parameters.</p>
<div class='horizontal-scroll'><table class="table table-striped table-hover table-responsive table" style="margin-left: auto; margin-right: auto; margin-left: auto; margin-right: auto;">
<caption>
<span id="tab:gsbymoransi">Table 1: </span>Significantly autocorrelated gene sets, ordered by decreasing Morans’I.
</caption>
<thead>
<tr>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
</th>
<th style="text-align:right;position: sticky; top:0; background-color: #FFFFFF;">
statistic
</th>
<th style="text-align:right;position: sticky; top:0; background-color: #FFFFFF;">
p.value
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
Mature.oligodendrocyte
</td>
<td style="text-align:right;">
0.8720535
</td>
<td style="text-align:right;">
0.0009990
</td>
</tr>
<tr>
<td style="text-align:left;">
Oligodendrocyte
</td>
<td style="text-align:right;">
0.8256448
</td>
<td style="text-align:right;">
0.0009990
</td>
</tr>
<tr>
<td style="text-align:left;">
Microglial.cell
</td>
<td style="text-align:right;">
0.6108444
</td>
<td style="text-align:right;">
0.0009990
</td>
</tr>
<tr>
<td style="text-align:left;">
Mid.radical.glial.cell
</td>
<td style="text-align:right;">
0.5321133
</td>
<td style="text-align:right;">
0.0009990
</td>
</tr>
<tr>
<td style="text-align:left;">
Astrocyte
</td>
<td style="text-align:right;">
0.5074616
</td>
<td style="text-align:right;">
0.0009990
</td>
</tr>
<tr>
<td style="text-align:left;">
Glial.cell
</td>
<td style="text-align:right;">
0.4630534
</td>
<td style="text-align:right;">
0.0009990
</td>
</tr>
<tr>
<td style="text-align:left;">
Lake.et.al.Science.Ex1
</td>
<td style="text-align:right;">
0.4003365
</td>
<td style="text-align:right;">
0.0009990
</td>
</tr>
<tr>
<td style="text-align:left;">
Oligodendrocyte.progenitor.cell
</td>
<td style="text-align:right;">
0.3873250
</td>
<td style="text-align:right;">
0.0009990
</td>
</tr>
<tr>
<td style="text-align:left;">
Neuron
</td>
<td style="text-align:right;">
0.3709289
</td>
<td style="text-align:right;">
0.0009990
</td>
</tr>
<tr>
<td style="text-align:left;">
Gene_SFARI_all
</td>
<td style="text-align:right;">
0.3504879
</td>
<td style="text-align:right;">
0.0009990
</td>
</tr>
<tr>
<td style="text-align:left;">
Oligodendrocyte.precursor.cell
</td>
<td style="text-align:right;">
0.3466464
</td>
<td style="text-align:right;">
0.0009990
</td>
</tr>
<tr>
<td style="text-align:left;">
Myeloid.cell
</td>
<td style="text-align:right;">
0.3020542
</td>
<td style="text-align:right;">
0.0009990
</td>
</tr>
<tr>
<td style="text-align:left;">
Gene_SFARI_high
</td>
<td style="text-align:right;">
0.2867298
</td>
<td style="text-align:right;">
0.0009990
</td>
</tr>
<tr>
<td style="text-align:left;">
Lake.et.al.Science.Ex8
</td>
<td style="text-align:right;">
0.2751603
</td>
<td style="text-align:right;">
0.0009990
</td>
</tr>
<tr>
<td style="text-align:left;">
Lake.et.al.Science.Ex6
</td>
<td style="text-align:right;">
0.2707421
</td>
<td style="text-align:right;">
0.0009990
</td>
</tr>
<tr>
<td style="text-align:left;">
Non.neuron
</td>
<td style="text-align:right;">
0.2679811
</td>
<td style="text-align:right;">
0.0009990
</td>
</tr>
<tr>
<td style="text-align:left;">
Lake.et.al.Science.Ex2
</td>
<td style="text-align:right;">
0.2506638
</td>
<td style="text-align:right;">
0.0009990
</td>
</tr>
<tr>
<td style="text-align:left;">
Inhibitory.neuron
</td>
<td style="text-align:right;">
0.2409752
</td>
<td style="text-align:right;">
0.0009990
</td>
</tr>
<tr>
<td style="text-align:left;">
Excitatory.neuron
</td>
<td style="text-align:right;">
0.2223157
</td>
<td style="text-align:right;">
0.0009990
</td>
</tr>
<tr>
<td style="text-align:left;">
Neural.progenitor.cell
</td>
<td style="text-align:right;">
0.2094519
</td>
<td style="text-align:right;">
0.0009990
</td>
</tr>
<tr>
<td style="text-align:left;">
Fibroblast
</td>
<td style="text-align:right;">
0.1974710
</td>
<td style="text-align:right;">
0.0009990
</td>
</tr>
<tr>
<td style="text-align:left;">
Glutamatergic.neuron
</td>
<td style="text-align:right;">
0.1897590
</td>
<td style="text-align:right;">
0.0009990
</td>
</tr>
<tr>
<td style="text-align:left;">
ASD.DATABASE
</td>
<td style="text-align:right;">
0.1803301
</td>
<td style="text-align:right;">
0.0009990
</td>
</tr>
<tr>
<td style="text-align:left;">
Gene_SFARI_syndromic
</td>
<td style="text-align:right;">
0.1636061
</td>
<td style="text-align:right;">
0.0009990
</td>
</tr>
<tr>
<td style="text-align:left;">
Radial.glial.cell
</td>
<td style="text-align:right;">
0.1630596
</td>
<td style="text-align:right;">
0.0009990
</td>
</tr>
<tr>
<td style="text-align:left;">
Truncated.radial.glial.cell
</td>
<td style="text-align:right;">
0.1499160
</td>
<td style="text-align:right;">
0.0009990
</td>
</tr>
<tr>
<td style="text-align:left;">
Progenitor.cell
</td>
<td style="text-align:right;">
0.1389278
</td>
<td style="text-align:right;">
0.0009990
</td>
</tr>
<tr>
<td style="text-align:left;">
Sensory.neuron
</td>
<td style="text-align:right;">
0.1313330
</td>
<td style="text-align:right;">
0.0009990
</td>
</tr>
<tr>
<td style="text-align:left;">
Deep.layer.neuron
</td>
<td style="text-align:right;">
0.1236193
</td>
<td style="text-align:right;">
0.0009990
</td>
</tr>
<tr>
<td style="text-align:left;">
Endothelial.cell
</td>
<td style="text-align:right;">
0.1176369
</td>
<td style="text-align:right;">
0.0009990
</td>
</tr>
<tr>
<td style="text-align:left;">
ID
</td>
<td style="text-align:right;">
0.1154954
</td>
<td style="text-align:right;">
0.0009990
</td>
</tr>
<tr>
<td style="text-align:left;">
Interneuron
</td>
<td style="text-align:right;">
0.1146196
</td>
<td style="text-align:right;">
0.0009990
</td>
</tr>
<tr>
<td style="text-align:left;">
ASD.CNV
</td>
<td style="text-align:right;">
0.1085017
</td>
<td style="text-align:right;">
0.0009990
</td>
</tr>
<tr>
<td style="text-align:left;">
SCZ.CNV
</td>
<td style="text-align:right;">
0.1059468
</td>
<td style="text-align:right;">
0.0009990
</td>
</tr>
<tr>
<td style="text-align:left;">
SCZ.Meta.analysis
</td>
<td style="text-align:right;">
0.1044286
</td>
<td style="text-align:right;">
0.0009990
</td>
</tr>
<tr>
<td style="text-align:left;">
Neural.stem.cell
</td>
<td style="text-align:right;">
0.1017215
</td>
<td style="text-align:right;">
0.0009990
</td>
</tr>
<tr>
<td style="text-align:left;">
Pericyte
</td>
<td style="text-align:right;">
0.0978981
</td>
<td style="text-align:right;">
0.0009990
</td>
</tr>
<tr>
<td style="text-align:left;">
Macrophage
</td>
<td style="text-align:right;">
0.0947877
</td>
<td style="text-align:right;">
0.0009990
</td>
</tr>
<tr>
<td style="text-align:left;">
Lake.et.al.Science.In3
</td>
<td style="text-align:right;">
0.0946480
</td>
<td style="text-align:right;">
0.0009990
</td>
</tr>
<tr>
<td style="text-align:left;">
Stem.cell
</td>
<td style="text-align:right;">
0.0929401
</td>
<td style="text-align:right;">
0.0009990
</td>
</tr>
<tr>
<td style="text-align:left;">
Neurodegenerative
</td>
<td style="text-align:right;">
0.0873586
</td>
<td style="text-align:right;">
0.0009990
</td>
</tr>
<tr>
<td style="text-align:left;">
Lake.et.al.Science.Ex4
</td>
<td style="text-align:right;">
0.0818042
</td>
<td style="text-align:right;">
0.0009990
</td>
</tr>
<tr>
<td style="text-align:left;">
SCZ.PGC.GWAS
</td>
<td style="text-align:right;">
0.0758406
</td>
<td style="text-align:right;">
0.0009990
</td>
</tr>
<tr>
<td style="text-align:left;">
SCZ.SNV
</td>
<td style="text-align:right;">
0.0678499
</td>
<td style="text-align:right;">
0.0009990
</td>
</tr>
<tr>
<td style="text-align:left;">
BPAD.GWAS
</td>
<td style="text-align:right;">
0.0445744
</td>
<td style="text-align:right;">
0.0009990
</td>
</tr>
<tr>
<td style="text-align:left;">
Epithelial.cell
</td>
<td style="text-align:right;">
0.0396025
</td>
<td style="text-align:right;">
0.0009990
</td>
</tr>
<tr>
<td style="text-align:left;">
Effector.memory.CD4..T.cell
</td>
<td style="text-align:right;">
0.0135817
</td>
<td style="text-align:right;">
0.0779221
</td>
</tr>
<tr>
<td style="text-align:left;">
Mesenchymal.stromal.cell
</td>
<td style="text-align:right;">
0.0131870
</td>
<td style="text-align:right;">
0.1178821
</td>
</tr>
<tr>
<td style="text-align:left;">
NDD
</td>
<td style="text-align:right;">
0.0116347
</td>
<td style="text-align:right;">
0.1068931
</td>
</tr>
<tr>
<td style="text-align:left;">
CD4..T.cell
</td>
<td style="text-align:right;">
0.0047427
</td>
<td style="text-align:right;">
0.3196803
</td>
</tr>
</tbody>
</table></div>
<p>We subset the GSVA ES calculated with default parameters to those gene sets
significantly autocorrelated with a Bonferroni FWER &lt; 0.01, and a minimum
Moran’s I value of 0.4.</p>
<pre class="r"><code>sig.genesets &lt;- rownames(moransi)[moransi$statistic &gt; 0.4]
spe.filt.es.sig &lt;- spe.filt.es[rownames(spe.filt.es) %in% sig.genesets]
dim(spe.filt.es.sig)
[1]    7 3591</code></pre>
</div>
</div>
<div id="cluster-annotation" class="section level1" number="5">
<h1><span class="header-section-number">5</span> Cluster annotation</h1>
<p>Here we use the previously selected autocorrelated gene sets to annotate
the gene-level clusters. We first define the following function for that
purpose, which takes as input an <code>SpatialExperiment</code> object with clusters
annotated in the <code>label</code> column metadata, and return a vector with as many
values as clusters and the most similar gene set for each cluster, in the
corresponding position.</p>
<pre class="r"><code>clustertogeneset &lt;- function(spe, p.value=0.01) {
  require(limma)

  y &lt;- assay(spe)
  f &lt;- colData(spe)$spatialLIBD
  stopifnot(is.factor(f) &amp;&amp; length(f) &gt; 1) ## QC
  nl &lt;- nlevels(f)
  mod &lt;- model.matrix(~ 0 + f)
  genesets &lt;- c()
  for (targetcluster in levels(f)) {
    fit &lt;- lmFit(y, mod)
    ## compare mean of the target cluster against the mean of
    ## all other clusters
    cf &lt;- sprintf(&quot;f%s=f%s-(%s)/%s&quot;, targetcluster, targetcluster,
                  paste(sprintf(&quot;f%s&quot;, setdiff(levels(f), targetcluster)),
                        collapse=&quot;+&quot;),
                  nl-1)
    contrast.matrix &lt;- makeContrasts(contrasts=cf, levels=mod)
    fit &lt;- contrasts.fit(fit, contrast.matrix)
    fit &lt;- eBayes(fit)
    tt &lt;- topTable(fit, coef=1, n=Inf)
    tt &lt;- tt[tt$adj.P.Val &lt; p.value, , drop=FALSE]
    genesets &lt;- c(genesets, rownames(tt)[1])
  }
  genesets
}</code></pre>
<p>Annotate gene sets to clusters and keep the unique ones using the GSVA ES
calculated with default parameters.</p>
<pre class="r"><code>annotatedgs &lt;- clustertogeneset(spe.filt.es.sig)
annotatedgs
[1] &quot;Astrocyte&quot;              &quot;Lake.et.al.Science.Ex1&quot; &quot;Oligodendrocyte&quot;       
[4] &quot;Oligodendrocyte&quot;        &quot;Mature.oligodendrocyte&quot; &quot;Lake.et.al.Science.Ex1&quot;
[7] &quot;Mature.oligodendrocyte&quot;</code></pre>
<pre class="r"><code>annotatedgs &lt;- unique(annotatedgs)
annotatedgs
[1] &quot;Astrocyte&quot;              &quot;Lake.et.al.Science.Ex1&quot; &quot;Oligodendrocyte&quot;       
[4] &quot;Mature.oligodendrocyte&quot;</code></pre>
<p>Figure <a href="#fig:hcannotclusters">5</a> shows these unique gene sets that mostly
correlated with some of the clusters.</p>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:hcannotclusters"></span>
<img src="04_gsva_hdlpfc_files/figure-html/hcannotclusters-1.png" alt="Annotation of clusters with gene sets autocorrelated with GSVA ES calculated with default parameters. (a) Original tissue image. (b) Spot annotation by gene-level clustering. (c) to (h) enrichment scores for each of the significantly autocorrelated gene sets that mostly correlated with at least one of the gene-level clusters." width="800px"  class="widefigure" />
<p class="caption">
Figure 5: <span class="caption-title">Annotation of clusters with gene sets autocorrelated with GSVA ES calculated with default parameters</span><br>(a) Original tissue image. (b) Spot annotation by gene-level clustering. (c) to (h) enrichment scores for each of the significantly autocorrelated gene sets that mostly correlated with at least one of the gene-level clusters.
</p>
</div>
<p>Finally, store the GSVA ES calculated with default parameters.</p>
<pre class="r"><code>saveRDS(spe.filt.es,
        file=file.path(&quot;_processed_data&quot;,
                       &quot;spe.filt.es.human_dlpfc.rds&quot;))</code></pre>
</div>
<div id="session-information" class="section level1" number="6">
<h1><span class="header-section-number">6</span> Session information</h1>
<pre class="r"><code>sessionInfo()
R version 4.4.0 (2024-04-24)
Platform: x86_64-pc-linux-gnu
Running under: Ubuntu 22.04.3 LTS

Matrix products: default
BLAS:   /usr/lib/x86_64-linux-gnu/openblas-pthread/libblas.so.3 
LAPACK: /usr/lib/x86_64-linux-gnu/openblas-pthread/libopenblasp-r0.3.20.so;  LAPACK version 3.10.0

locale:
 [1] LC_CTYPE=ca_ES.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=ca_ES.UTF-8        LC_COLLATE=ca_ES.UTF-8    
 [5] LC_MONETARY=ca_ES.UTF-8    LC_MESSAGES=ca_ES.UTF-8   
 [7] LC_PAPER=ca_ES.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=ca_ES.UTF-8 LC_IDENTIFICATION=C       

time zone: Europe/Madrid
tzcode source: system (glibc)

attached base packages:
[1] stats4    stats     graphics  grDevices utils     datasets  methods  
[8] base     

other attached packages:
 [1] limma_3.60.2                   Voyager_1.6.0                 
 [3] SpatialFeatureExperiment_1.6.1 nnSVG_1.8.0                   
 [5] BiocParallel_1.38.0            GSVA_1.52.2                   
 [7] RSpectra_0.16-1                SpatialPCA_1.3.0              
 [9] ggspavis_1.10.0                patchwork_1.2.0               
[11] RColorBrewer_1.1-3             scater_1.32.0                 
[13] ggplot2_3.5.1                  scran_1.32.0                  
[15] scuttle_1.14.0                 SpatialExperiment_1.14.0      
[17] SingleCellExperiment_1.26.0    SummarizedExperiment_1.34.0   
[19] Biobase_2.64.0                 GenomicRanges_1.56.0          
[21] GenomeInfoDb_1.40.1            IRanges_2.38.0                
[23] S4Vectors_0.42.0               BiocGenerics_0.50.0           
[25] MatrixGenerics_1.16.0          matrixStats_1.3.0             
[27] kableExtra_1.4.0               knitr_1.47                    
[29] BiocStyle_2.32.0              

loaded via a namespace (and not attached):
  [1] R.methodsS3_1.8.2         GSEABase_1.66.0          
  [3] tiff_0.1-12               goftest_1.2-3            
  [5] Biostrings_2.72.0         HDF5Array_1.32.0         
  [7] vctrs_0.6.5               spatstat.random_3.2-3    
  [9] digest_0.6.35             png_0.1-8                
 [11] proxy_0.4-27              ggrepel_0.9.5            
 [13] deldir_2.0-4              parallelly_1.37.1        
 [15] magick_2.8.3              MASS_7.3-60.2            
 [17] reshape2_1.4.4            scico_1.5.0              
 [19] httpuv_1.6.15             foreach_1.5.2            
 [21] withr_3.0.0               xfun_0.44                
 [23] survival_3.6-4            memoise_2.0.1            
 [25] s2_1.1.6                  ggbeeswarm_0.7.2         
 [27] systemfonts_1.1.0         zoo_1.8-12               
 [29] pbapply_1.7-2             R.oo_1.26.0              
 [31] spData_2.3.1              KEGGREST_1.44.0          
 [33] promises_1.3.0            SPARK_1.1.1              
 [35] httr_1.4.7                globals_0.16.3           
 [37] fitdistrplus_1.1-11       rhdf5filters_1.16.0      
 [39] rhdf5_2.48.0              rstudioapi_0.16.0        
 [41] UCSC.utils_1.0.0          units_0.8-5              
 [43] miniUI_0.1.1.1            generics_0.1.3           
 [45] curl_5.2.1                zlibbioc_1.50.0          
 [47] ScaledMatrix_1.12.0       polyclip_1.10-6          
 [49] GenomeInfoDbData_1.2.12   SparseArray_1.4.8        
 [51] fftwtools_0.9-11          xtable_1.8-4             
 [53] stringr_1.5.1             pracma_2.4.4             
 [55] doParallel_1.0.17         evaluate_0.23            
 [57] S4Arrays_1.4.1            BiocFileCache_2.12.0     
 [59] bookdown_0.39             irlba_2.3.5.1            
 [61] colorspace_2.1-0          filelock_1.0.3           
 [63] ROCR_1.0-11               reticulate_1.37.0        
 [65] spatstat.data_3.0-4       magrittr_2.0.3           
 [67] lmtest_0.9-40             spdep_1.3-4              
 [69] later_1.3.2               viridis_0.6.5            
 [71] lattice_0.22-6            spatstat.geom_3.2-9      
 [73] future.apply_1.11.2       scattermore_1.2          
 [75] XML_3.99-0.16.1           cowplot_1.1.3            
 [77] RcppAnnoy_0.0.22          class_7.3-22             
 [79] pillar_1.9.0              nlme_3.1-164             
 [81] iterators_1.0.14          EBImage_4.46.0           
 [83] compiler_4.4.0            beachmat_2.20.0          
 [85] stringi_1.8.4             sf_1.0-16                
 [87] tensor_1.5                plyr_1.8.9               
 [89] crayon_1.5.2              abind_1.4-5              
 [91] locfit_1.5-9.9            sp_2.1-4                 
 [93] terra_1.7-78              bit_4.0.5                
 [95] dplyr_1.1.4               codetools_0.2-20         
 [97] BiocSingular_1.20.0       openssl_2.2.0            
 [99] bslib_0.7.0               e1071_1.7-14             
[101] plotly_4.10.4             mime_0.12                
[103] splines_4.4.0             Rcpp_1.0.12              
[105] fastDummies_1.7.3         dbplyr_2.5.0             
[107] sparseMatrixStats_1.16.0  blob_1.2.4               
[109] utf8_1.2.4                listenv_0.9.1            
[111] DelayedMatrixStats_1.26.0 rdist_0.0.5              
[113] tibble_3.2.1              Matrix_1.7-0             
[115] statmod_1.5.0             svglite_2.1.3            
[117] pkgconfig_2.0.3           tools_4.4.0              
[119] BRISC_1.0.5               cachem_1.1.0             
[121] RSQLite_2.3.7             viridisLite_0.4.2        
[123] DBI_1.2.2                 fastmap_1.2.0            
[125] rmarkdown_2.27            scales_1.3.0             
[127] grid_4.4.0                ica_1.0-3                
[129] Seurat_5.1.0              sass_0.4.9               
[131] BiocManager_1.30.23       dotCall64_1.1-1          
[133] graph_1.82.0              RANN_2.6.1               
[135] farver_2.1.2              wk_0.9.1                 
[137] yaml_2.3.8                cli_3.6.2                
[139] purrr_1.0.2               leiden_0.4.3.1           
[141] lifecycle_1.0.4           askpass_1.2.0            
[143] uwot_0.2.2                bluster_1.14.0           
[145] DropletUtils_1.24.0       annotate_1.82.0          
[147] gtable_0.3.5              rjson_0.2.21             
[149] umap_0.2.10.0             ggridges_0.5.6           
[151] progressr_0.14.0          parallel_4.4.0           
[153] jsonlite_1.8.8            edgeR_4.2.0              
[155] RcppHNSW_0.6.0            bitops_1.0-7             
[157] bit64_4.0.5               Rtsne_0.17               
[159] spatstat.utils_3.0-4      BiocNeighbors_1.22.0     
[161] SeuratObject_5.0.2        matlab_1.0.4             
[163] ggside_0.3.1              jquerylib_0.1.4          
[165] highr_0.11                metapod_1.12.0           
[167] dqrng_0.4.1               zeallot_0.1.0            
[169] R.utils_2.12.3            lazyeval_0.2.2           
[171] shiny_1.8.1.1             htmltools_0.5.8.1        
[173] sctransform_0.4.1         tinytex_0.51             
[175] glue_1.7.0                spam_2.10-0              
[177] XVector_0.44.0            RCurl_1.98-1.14          
[179] classInt_0.4-10           jpeg_0.1-10              
[181] gridExtra_2.3             boot_1.3-30              
[183] igraph_2.0.3              R6_2.5.1                 
[185] tidyr_1.3.1               CompQuadForm_1.4.3       
[187] labeling_0.4.3            cluster_2.1.6            
[189] Rhdf5lib_1.26.0           memuse_4.2-3             
[191] DelayedArray_0.30.1       tidyselect_1.2.1         
[193] vipor_0.4.7               xml2_1.3.6               
[195] AnnotationDbi_1.66.0      future_1.33.2            
[197] sfheaders_0.4.4           rsvd_1.0.5               
[199] munsell_0.5.1             KernSmooth_2.23-24       
[201] data.table_1.15.4         htmlwidgets_1.6.4        
[203] rlang_1.1.3               spatstat.sparse_3.0-3    
[205] spatstat.explore_3.2-7    pdist_1.2.1              
[207] ggnewscale_0.4.10         fansi_1.0.6              
[209] beeswarm_0.4.0           </code></pre>
</div>
<div id="references" class="section level1" number="7">
<h1><span class="header-section-number">7</span> References</h1>
</div>



</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": {
      styles: {
        ".MathJax_Display": {
           "text-align": "center",
           padding: "0px 150px 0px 65px",
           margin: "0px 0px 0.5em"
        },
        "@media screen and (max-width: 991px)": {
            ".MathJax_Display": {
               "text-align": "center",
               padding: "0 0 0 0"
            }
         }
      }
    }
  });
</script>
<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

<script type="text/javascript">
$(document).ready(function ()  {
  
  // Map "enter" keypress to the same action as a cursor click
  function navigateLink(e) {
    if (e.key === "Enter") {
      $(this).trigger("click");
    }
  }

  var toc_items = document.querySelectorAll(".tocify-item");
  for (var i = 0; i < toc_items.length; i++) {
    // The link role tells screen readers this is for navigation
    toc_items.item(i).setAttribute("role", "link");
    // tabindex = 0 allows selection via keyboard tab presses
    toc_items.item(i).setAttribute("tabindex", "0");
    // Listen for "Enter" keypress when item is selected
    toc_items.item(i).addEventListener("keydown", navigateLink);
  }
});
</script>

</body>
</html>
