<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>GSVA analysis mouse brain data</title>

<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       </style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/default.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
  p.abstract{
    text-align: center;
    font-weight: bold;
  }
  div.abstract{
    margin: auto;
    width: 90%;
  }
</style>



<link rel="stylesheet" href="/home/peremoles/R/x86_64-pc-linux-gnu-library/4.4/BiocStyle/resources/html/bioconductor.css" type="text/css" />



<style type = "text/css">
.main-container {
  max-width: 828px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {

}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 246px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



<script>
function toggle_visibility(id1) {
  var e = document.getElementById(id1);
  e.style.display = ((e.style.display!="none") ? "none" : "block");
}
</script>

</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">GSVA analysis of spatial transcriptomics data</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="01_qa_pr_mouse.html">QA &amp; processing mouse brain data</a>
</li>
<li>
  <a href="02_gsva_hc.html">GSVA analysis mouse brain data</a>
</li>
<li>
  <a href="03_qa_pr_hdlpfc.html">QA &amp; processing human DLPFC data</a>
</li>
<li>
  <a href="04_gsva_hdlpfc.html">GSVA analysis human DLPFC data</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">GSVA analysis mouse brain data</h1>
<p class="author-name">Pere Moles<span class="affil-mark">1*</span></p>
<p class="author-affiliation"><span class="affil-mark">1</span>Dept. of Medicine and Life Sciences, Universitat Pompeu Fabra</p>
<p class="author-email"><span class="affil-mark">*</span><a href="mailto:pere.moles@upf.edu">pere.moles@upf.edu</a></p>
<h4 class="date">de maig 30, 2024</h4>
<h4 class="abstract">Abstract</h4>
<p>Here we perform a GSVA analysis of the 10x Visium mouse brain data.</p>

</div>


<div id="download-data" class="section level1" number="1">
<h1><span class="header-section-number">1</span> Download data</h1>
<p>We start by downloading the data, following the instructions at
<a href="http://research.libd.org/spatialLIBD/articles/spatialLIBD.html" class="uri">http://research.libd.org/spatialLIBD/articles/spatialLIBD.html</a></p>
<pre class="r"><code>
library(TENxVisiumData)
library(SpatialExperiment)

spe &lt;- MouseBrainSagittalAnterior()
spe &lt;- spe[, colData(spe)$sample_id == &quot;MouseBrainSagittalAnterior1&quot;]</code></pre>
<p>This is the row data.</p>
<pre class="r"><code>rowData(spe)
DataFrame with 32285 rows and 1 column
                        symbol
                   &lt;character&gt;
ENSMUSG00000051951        Xkr4
ENSMUSG00000089699      Gm1992
ENSMUSG00000102331     Gm19938
ENSMUSG00000102343     Gm37381
ENSMUSG00000025900         Rp1
...                        ...
ENSMUSG00000095523  AC124606.1
ENSMUSG00000095475  AC133095.2
ENSMUSG00000094855  AC133095.1
ENSMUSG00000095019  AC234645.1
ENSMUSG00000095041  AC149090.1</code></pre>
<p>This is the colum data. We have multiple samples and a few thousand spots per
sample.</p>
<pre class="r"><code>colData(spe)
DataFrame with 2695 rows and 1 column
                                sample_id
                              &lt;character&gt;
AAACAAGTATCTCCCA-1 MouseBrainSagittalAn..
AAACACCAATAACTGC-1 MouseBrainSagittalAn..
AAACAGAGCGACTCCT-1 MouseBrainSagittalAn..
AAACAGCTTTCAGAAG-1 MouseBrainSagittalAn..
AAACAGGGTCTATATT-1 MouseBrainSagittalAn..
...                                   ...
TTGTGTTTCCCGAAAG-1 MouseBrainSagittalAn..
TTGTTCAGTGTGCTAC-1 MouseBrainSagittalAn..
TTGTTGTGTGTCAAGA-1 MouseBrainSagittalAn..
TTGTTTCACATCCAGG-1 MouseBrainSagittalAn..
TTGTTTCCATACAACT-1 MouseBrainSagittalAn..</code></pre>
<pre class="r"><code>table(colData(spe)$sample_id)

MouseBrainSagittalAnterior1 
                       2695 </code></pre>
<p>For simplicity, we are going to analyze here only the sample with
identifier <code>MouseBrainSagittalAnterior1</code>.</p>
<pre class="r"><code>spe &lt;- spe[, colData(spe)$sample_id %in% &quot;MouseBrainSagittalAnterior1&quot;]
dim(spe)
[1] 32285  2695</code></pre>
<p>There is an indicator column called <code>in_tissue</code> for identifying spots
overlapping the tissue, but apparently all spots have such a condition.</p>
<pre class="r"><code>stopifnot(all(colData(spe)$in_tissue))</code></pre>
<p>This is the internal column metadata.</p>
<pre class="r"><code>int_colData(spe)
DataFrame with 2695 rows and 5 columns
     reducedDims     altExps    colPairs spatialCoords spatialData
     &lt;DataFrame&gt; &lt;DataFrame&gt; &lt;DataFrame&gt;      &lt;matrix&gt; &lt;DataFrame&gt;
1                                            7474:8500 TRUE:50:102
2                                            8552:2788  TRUE:59:19
3                                            3163:7950  TRUE:14:94
4                                            6636:2100   TRUE:43:9
5                                            7115:2375  TRUE:47:13
...          ...         ...         ...           ...         ...
2691                                         7594:5541  TRUE:51:59
2692                                         4361:5885  TRUE:24:64
2693                                         5199:6780  TRUE:31:77
2694                                         8433:4371  TRUE:58:42
2695                                         6876:3339  TRUE:45:27</code></pre>
</div>
<div id="quality-assessment" class="section level1" number="2">
<h1><span class="header-section-number">2</span> Quality assessment</h1>
<p>Identify mitochondrial (MT) genes, here the column with the gene HUGO symbols
is called <code>gene_name</code>.</p>
<pre class="r"><code>is_mito &lt;- grepl(&quot;(^MT-)|(^mt-)&quot;, rowData(spe)$gene_name)
table(is_mito)
&lt; table of extent 0 &gt;</code></pre>
<pre class="r"><code>rowData(spe)$gene_name[is_mito]
NULL</code></pre>
<p>Add quality control metrics.</p>
<pre class="r"><code>library(scuttle)
library(ggspavis)
library(patchwork)

spe &lt;- addPerCellQC(spe, subsets=list(mito=is_mito))</code></pre>
<p>Explore sequencing depth in UMI counts per spot. Figure
<a href="#fig:mousetissueUMIcntxspot">1</a> shows in the tissue image the spatial
pattern of sequencing depth in UMI counts per spot.</p>
<pre class="r"><code>## image seems to be reversed and for this reason we set x_coord=&quot;pxl_col_in_fullres&quot;
## and y_coord=&quot;pxl_row_in_fullres&quot;
plotVisium(spe, x_coord=&quot;pxl_row_in_fullres&quot;,
           y_coord=&quot;pxl_col_in_fullres&quot;,
           annotate=&quot;sum&quot;, point_size=1,
           pal=c(&quot;skyblue&quot;, &quot;darkred&quot;), facets=NULL)</code></pre>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:mousetissueUMIcntxspot"></span>
<img src="mouse_files/figure-html/mousetissueUMIcntxspot-1.png" alt="Spatial pattern of sequencing depth per spot in UMI counts." width="600px"  class="widefigure" />
<p class="caption">
Figure 1: <span class="caption-title">Spatial pattern of sequencing depth per spot in UMI counts</span><br>
</p>
</div>
<p>Figure <a href="#fig:mouseUMIcntxspot">2</a>a
shows the frequency distribution of UMI counts per spot and panels (b) to (d)
show the spatial distribution of discarded spots at different cutoffs of
sequencing depth (2000, 1000 and 700).</p>
<pre class="r"><code>hp &lt;- ggplot(data=colData(spe), aes(x=sum)) +
        geom_histogram(bins=40) +
        labs(x=&quot;UMI counts per spot&quot;, y=&quot;Frequency&quot;, tag=&quot;a&quot;) +
        geom_vline(aes(xintercept=2000), col=&quot;darkgreen&quot;) +
        geom_vline(aes(xintercept=1000), col=&quot;darkorange&quot;) +
        geom_vline(aes(xintercept=700), col=&quot;darkred&quot;)
colData(spe)$qc_lib_size &lt;- colData(spe)$sum &lt; 2000
qp2k &lt;- plotSpotQC(spe, plot_type=&quot;spot&quot;, annotate=&quot;qc_lib_size&quot;) +
          theme(legend.position=&quot;none&quot;) + labs(tag=&quot;b&quot;)
colData(spe)$qc_lib_size &lt;- colData(spe)$sum &lt; 1000
qp1k &lt;- plotSpotQC(spe, plot_type=&quot;spot&quot;, annotate=&quot;qc_lib_size&quot;) +
          theme(legend.position=&quot;none&quot;) + labs(tag=&quot;c&quot;)
colData(spe)$qc_lib_size &lt;- colData(spe)$sum &lt; 700
qp700 &lt;- plotSpotQC(spe, plot_type=&quot;spot&quot;, annotate=&quot;qc_lib_size&quot;) +
          theme(legend.position=&quot;none&quot;) + labs(tag=&quot;d&quot;)
(hp | qp2k) / (qp1k | qp700)</code></pre>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:mouseUMIcntxspot"></span>
<img src="mouse_files/figure-html/mouseUMIcntxspot-1.png" alt="Sequencing depth per spot in UMI counts. (a) Frequency distribution of UMI counts, where vertical lines indicate possible cutoff UMI count values below which spots would be discarded. (b) to (d) spatial distribution of spots discarded using the ppreviously considered sequencing depth cutoff values." width="800px"  class="widefigure" />
<p class="caption">
Figure 2: <span class="caption-title">Sequencing depth per spot in UMI counts</span><br>(a) Frequency distribution of UMI counts, where vertical lines indicate possible cutoff UMI count values below which spots would be discarded. (b) to (d) spatial distribution of spots discarded using the ppreviously considered sequencing depth cutoff values.
</p>
</div>
<p>Explore number of genes with non-zero UMI counts per spot. Figure
<a href="#fig:mousenonzeroUMIcntxspot">3</a>a shows the frequency distribution of genes
per spot with non-zero UMI counts, while panels (b) do (d)
show the spatial distribution of discarded spots at different cutoffs of
sequencing depth (1000, 750 and 500).</p>
<pre class="r"><code>hp &lt;- ggplot(data=colData(spe), aes(x=detected)) +
        geom_histogram(bins=40) +
        labs(x=&quot;Genes with non-zero UMI counts per spot&quot;,
             y=&quot;Frequency&quot;, tag=&quot;a&quot;) +
        geom_vline(aes(xintercept=1000), col=&quot;darkgreen&quot;) +
        geom_vline(aes(xintercept=750), col=&quot;darkorange&quot;) +
        geom_vline(aes(xintercept=500), col=&quot;darkred&quot;)
colData(spe)$qc_detected &lt;- colData(spe)$detected &lt; 1000
qp1k &lt;- plotSpotQC(spe, plot_type=&quot;spot&quot;, annotate=&quot;qc_detected&quot;) +
          theme(legend.position=&quot;none&quot;) + labs(tag=&quot;b&quot;)
colData(spe)$qc_detected &lt;- colData(spe)$detected &lt; 750
qp750 &lt;- plotSpotQC(spe, plot_type=&quot;spot&quot;, annotate=&quot;qc_detected&quot;) +
          theme(legend.position=&quot;none&quot;) + labs(tag=&quot;c&quot;)
colData(spe)$qc_detected &lt;- colData(spe)$detected &lt; 500
qp500 &lt;- plotSpotQC(spe, plot_type=&quot;spot&quot;, annotate=&quot;qc_detected&quot;) +
          theme(legend.position=&quot;none&quot;) + labs(tag=&quot;d&quot;)
(hp | qp1k) / (qp750 | qp500)</code></pre>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:mousenonzeroUMIcntxspot"></span>
<img src="mouse_files/figure-html/mousenonzeroUMIcntxspot-1.png" alt="Non-zero UMI counts per spot. (a) Frequency distribution of non-zero UMI counts per spot, where vertical lines indicate possible cutoff values in the number of genes with non-zero UMI counts, below which spots would be discarded. (b) to (d) spatial distribution of spots discarded using the ppreviously considered sequencing depth cutoff values." width="800px"  class="widefigure" />
<p class="caption">
Figure 3: <span class="caption-title">Non-zero UMI counts per spot</span><br>(a) Frequency distribution of non-zero UMI counts per spot, where vertical lines indicate possible cutoff values in the number of genes with non-zero UMI counts, below which spots would be discarded. (b) to (d) spatial distribution of spots discarded using the ppreviously considered sequencing depth cutoff values.
</p>
</div>
<p>Figure <a href="#fig:mouseMTUMIcntxspot">4</a> shows the proportion of MT UMI counts per
spot.</p>
<pre class="r"><code>hp &lt;- ggplot(data=colData(spe), aes(x=subsets_mito_percent)) +
        geom_histogram(bins=40) +
        labs(x=&quot;Proportion of mitochondrial UMI counts per spot&quot;,
             y=&quot;Frequency&quot;, tag=&quot;a&quot;) +
        geom_vline(aes(xintercept=25), col=&quot;darkgreen&quot;) +
        geom_vline(aes(xintercept=27), col=&quot;darkorange&quot;) +
        geom_vline(aes(xintercept=29), col=&quot;darkred&quot;)
colData(spe)$qc_mito &lt;- colData(spe)$subsets_mito_percent &gt; 25
qp25 &lt;- plotSpotQC(spe, plot_type=&quot;spot&quot;, annotate=&quot;qc_mito&quot;) +
          theme(legend.position=&quot;none&quot;) + labs(tag=&quot;b&quot;)
colData(spe)$qc_mito &lt;- colData(spe)$subsets_mito_percent &gt; 27
qp27 &lt;- plotSpotQC(spe, plot_type=&quot;spot&quot;, annotate=&quot;qc_mito&quot;) +
          theme(legend.position=&quot;none&quot;) + labs(tag=&quot;c&quot;)
colData(spe)$qc_mito &lt;- colData(spe)$subsets_mito_percent &gt; 29
qp29 &lt;- plotSpotQC(spe, plot_type=&quot;spot&quot;, annotate=&quot;qc_mito&quot;) +
          theme(legend.position=&quot;none&quot;) + labs(tag=&quot;d&quot;)
(hp | qp25) / (qp27 | qp29)</code></pre>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:mouseMTUMIcntxspot"></span>
<img src="mouse_files/figure-html/mouseMTUMIcntxspot-1.png" alt="Mitochondrial UMI counts per spot. (a) Frequency distribution of the proportion of mitochondrial UMI counts per spot, where vertical lines indicate possible cutoff values in the proportion of UMI counts from mitochondrial genes, above which spots would be discarded. (b) to (d) spatial distribution of spots discarded using the ppreviously considered sequencing depth cutoff values." width="800px"  class="widefigure" />
<p class="caption">
Figure 4: <span class="caption-title">Mitochondrial UMI counts per spot</span><br>(a) Frequency distribution of the proportion of mitochondrial UMI counts per spot, where vertical lines indicate possible cutoff values in the proportion of UMI counts from mitochondrial genes, above which spots would be discarded. (b) to (d) spatial distribution of spots discarded using the ppreviously considered sequencing depth cutoff values.
</p>
</div>
<p>We use the more lenient cutoff values in the previous three
QC metrics, which lead to the following amount of discarded
spots per metric.</p>
<pre class="r"><code>colData(spe)$discard &lt;- colData(spe)$qc_lib_size |
                        colData(spe)$qc_detected |
                        colData(spe)$qc_mito
table(colData(spe)$discard)

FALSE  TRUE 
 2690     5 </code></pre>
<p>Figure <a href="#fig:mousediscardedspots">5</a> below shows the spatial pattern of the
discarded spots.</p>
<pre class="r"><code>pqc &lt;- plotSpotQC(spe, plot_type=&quot;spot&quot;, annotate=&quot;discard&quot;) +
         theme(legend.position=&quot;none&quot;) + labs(tag=&quot;a&quot;)
## image seems to be reversed and for this reason we set x_coord=&quot;pxl_col_in_fullres&quot;
## and y_coord=&quot;pxl_row_in_fullres&quot;
pts &lt;- plotVisium(spe, x_coord=&quot;pxl_row_in_fullres&quot;, facets=NULL,
                  y_coord=&quot;pxl_col_in_fullres&quot;, annotate=&quot;discard&quot;) +
         labs(tag=&quot;b&quot;) + theme(legend.position=&quot;none&quot;)
(pqc | pts)</code></pre>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:mousediscardedspots"></span>
<img src="mouse_files/figure-html/mousediscardedspots-1.png" alt="Spatial pattern of discarded spots. (a) Discarded spots highlighted in red in the spot grid. (b) Discarded spots highlighted in blue in the tissue plot." width="800px"  class="widefigure" />
<p class="caption">
Figure 5: <span class="caption-title">Spatial pattern of discarded spots</span><br>(a) Discarded spots highlighted in red in the spot grid. (b) Discarded spots highlighted in blue in the tissue plot.
</p>
</div>
<p>Finally, we filter out these low-quality spots.</p>
<pre class="r"><code>dim(spe)
[1] 32285  2695</code></pre>
<pre class="r"><code>spe &lt;- spe[, !colData(spe)$discard]
dim(spe)
[1] 32285  2690</code></pre>
</div>
<div id="filter-lowly-expressed-genes" class="section level1" number="3">
<h1><span class="header-section-number">3</span> Filter lowly-expressed genes</h1>
<p>Figure <a href="#fig:mouseUMIcntxgene">6</a> shows the empirical distribution of UMI
counts per gene.</p>
<pre class="r"><code>cntxgene &lt;- rowSums(assays(spe)$counts)+1
plot.ecdf(cntxgene, xaxt=&quot;n&quot;, panel.first=grid(), xlab=&quot;UMI counts per gene&quot;,
          las=1, log=&quot;x&quot;, xlim=c(1, 10e5), main=&quot;&quot;)
axis(1, at=10^(0:5), labels=10^(0:5))
abline(v=100, lwd=2, col=&quot;red&quot;)</code></pre>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:mouseUMIcntxgene"></span>
<img src="mouse_files/figure-html/mouseUMIcntxgene-1.png" alt="UMI counts per gene. The vertical red line shows a cutoff value below which we discard lowly-expressed genes." width="500px"  class="widefigure" />
<p class="caption">
Figure 6: <span class="caption-title">UMI counts per gene</span><br>The vertical red line shows a cutoff value below which we discard lowly-expressed genes.
</p>
</div>
<p>We filter out genes with less than 100 counts throughout all spots.</p>
<pre class="r"><code>dim(spe)
[1] 32285  2690</code></pre>
<pre class="r"><code>spe.filt &lt;- spe[cntxgene &gt; 100, ]
dim(spe.filt)
[1] 13469  2690</code></pre>
</div>
<div id="normalization" class="section level1" number="4">
<h1><span class="header-section-number">4</span> Normalization</h1>
<p>Calculate normalization factors and normalized expression values in logarithmic
scale.</p>
<pre class="r"><code>spe.filt &lt;- computeLibraryFactors(spe.filt)
summary(sizeFactors(spe.filt))
   Min. 1st Qu.  Median    Mean 3rd Qu.    Max. 
0.03655 0.70188 1.00850 1.00000 1.29545 2.76666 </code></pre>
<pre class="r"><code>spe.filt &lt;- logNormCounts(spe.filt)
assayNames(spe.filt)
[1] &quot;counts&quot;    &quot;logcounts&quot;</code></pre>
<pre class="r"><code>assays(spe.filt)$logcounts[1:5, 1:4]
5 x 4 sparse Matrix of class &quot;dgCMatrix&quot;
                   AAACAAGTATCTCCCA-1 AAACACCAATAACTGC-1 AAACAGAGCGACTCCT-1
ENSMUSG00000025902           .                 0.7208732          .        
ENSMUSG00000033845           2.227128          0.7208732          2.1493081
ENSMUSG00000025903           .                 0.7208732          1.4425853
ENSMUSG00000033813           2.227128          1.1993479          0.8945559
ENSMUSG00000002459           .                 0.7208732          .        
                   AAACAGCTTTCAGAAG-1
ENSMUSG00000025902           .       
ENSMUSG00000033845           .       
ENSMUSG00000025903           1.148245
ENSMUSG00000033813           1.498078
ENSMUSG00000002459           1.498078</code></pre>
</div>
<div id="dimensionality-reduction" class="section level1" number="5">
<h1><span class="header-section-number">5</span> Dimensionality reduction</h1>
<p>Select a subset of top highly-variable genes (HVGs).</p>
<pre class="r"><code>library(scran)

dec &lt;- modelGeneVar(spe.filt, assay.type=&quot;logcounts&quot;)</code></pre>
<p>Figure <a href="#fig:mousemeanvar">7</a> shows the mean-variance relationship.</p>
<pre class="r"><code>fit &lt;- metadata(dec)
plot(fit$mean, fit$var, xlab=&quot;mean of log-expression&quot;,
     ylab=&quot;variance of log-expression&quot;)
curve(fit$trend(x), col=&quot;dodgerblue&quot;, add=TRUE, lwd=2)</code></pre>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:mousemeanvar"></span>
<img src="mouse_files/figure-html/mousemeanvar-1.png" alt="Mean-variance relationship." width="500px"  class="widefigure" />
<p class="caption">
Figure 7: <span class="caption-title">Mean-variance relationship</span><br>
</p>
</div>
<p>Select top 10% HVGs.</p>
<pre class="r"><code>top_hvgs &lt;- getTopHVGs(dec, prop=0.1)
length(top_hvgs)
[1] 597</code></pre>
<p>Calculate PCA.</p>
<pre class="r"><code>library(scater)

set.seed(123)
spe.filt &lt;- runPCA(spe.filt, subset_row=top_hvgs)

reducedDimNames(spe.filt)
[1] &quot;PCA&quot;</code></pre>
<pre class="r"><code>dim(reducedDim(spe.filt, &quot;PCA&quot;))
[1] 2690   50</code></pre>
</div>
<div id="clustering" class="section level1" number="6">
<h1><span class="header-section-number">6</span> Clustering</h1>
<p>Here we perform a non-spatial graph-based clustering of the spots,
targeting 7 clusters, which is the number of anatomically annotated
groups of spots (six cortical layers and white matter) stored in the
column <code>spatialLIBD</code>.</p>
<pre class="r"><code>table(colData(spe.filt)$spatialLIBD)
&lt; table of extent 0 &gt;</code></pre>
<pre class="r"><code>k &lt;- 30
g &lt;- buildSNNGraph(spe.filt, k=k, use.dimred=&quot;PCA&quot;)
set.seed(123)
g_walk &lt;- igraph::cluster_walktrap(g)
colLabels(spe.filt) &lt;- factor(g_walk$membership)
table(colLabels(spe.filt))

  1   2   3   4   5   6   7   8   9 
355 260 383 548 185 555 155 116 133 </code></pre>
<p>Figure <a href="#fig:mouseclustering">8</a> shows the clustering of the spots.</p>
<pre class="r"><code>library(RColorBrewer)
library(patchwork)

set.seed(123)
pltcl &lt;- plotVisium(spe.filt, x_coord=&quot;pxl_row_in_fullres&quot;,
                    y_coord=&quot;pxl_col_in_fullres&quot;, point_size=1, annotate=&quot;label&quot;,
                    pal=brewer.pal(nlevels(colLabels(spe.filt)), &quot;Set1&quot;))
pltcl</code></pre>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:mouseclustering"></span>
<img src="mouse_files/figure-html/mouseclustering-1.png" alt="Clustering of the spots." width="800px"  class="widefigure" />
<p class="caption">
Figure 8: <span class="caption-title">Clustering of the spots</span><br>
</p>
</div>
<p>Finally, store the quality controlled, normalized and processed data.</p>
<pre class="r"><code>saveRDS(spe.filt, file=file.path(&quot;_processed_data&quot;,
                                 &quot;spe.filt.mouse_brain.rds&quot;))</code></pre>
</div>
<div id="session-information" class="section level1" number="7">
<h1><span class="header-section-number">7</span> Session information</h1>
<pre class="r"><code>sessionInfo()
R version 4.4.0 (2024-04-24)
Platform: x86_64-pc-linux-gnu
Running under: Ubuntu 22.04.3 LTS

Matrix products: default
BLAS:   /usr/lib/x86_64-linux-gnu/openblas-pthread/libblas.so.3 
LAPACK: /usr/lib/x86_64-linux-gnu/openblas-pthread/libopenblasp-r0.3.20.so;  LAPACK version 3.10.0

locale:
 [1] LC_CTYPE=ca_ES.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=ca_ES.UTF-8        LC_COLLATE=ca_ES.UTF-8    
 [5] LC_MONETARY=ca_ES.UTF-8    LC_MESSAGES=ca_ES.UTF-8   
 [7] LC_PAPER=ca_ES.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=ca_ES.UTF-8 LC_IDENTIFICATION=C       

time zone: Europe/Madrid
tzcode source: system (glibc)

attached base packages:
[1] stats4    stats     graphics  grDevices utils     datasets  methods  
[8] base     

other attached packages:
 [1] RColorBrewer_1.1-3          scater_1.32.0              
 [3] scran_1.32.0                patchwork_1.2.0            
 [5] ggspavis_1.10.0             ggplot2_3.5.1              
 [7] scuttle_1.14.0              TENxVisiumData_1.12.0      
 [9] SpatialExperiment_1.14.0    SingleCellExperiment_1.26.0
[11] SummarizedExperiment_1.34.0 Biobase_2.64.0             
[13] GenomicRanges_1.56.0        GenomeInfoDb_1.40.1        
[15] IRanges_2.38.0              S4Vectors_0.42.0           
[17] MatrixGenerics_1.16.0       matrixStats_1.3.0          
[19] ExperimentHub_2.12.0        AnnotationHub_3.12.0       
[21] BiocFileCache_2.12.0        dbplyr_2.5.0               
[23] BiocGenerics_0.50.0         kableExtra_1.4.0           
[25] knitr_1.47                  BiocStyle_2.32.0           

loaded via a namespace (and not attached):
  [1] rstudioapi_0.16.0         jsonlite_1.8.8           
  [3] magrittr_2.0.3            ggbeeswarm_0.7.2         
  [5] magick_2.8.3              farver_2.1.2             
  [7] rmarkdown_2.27            zlibbioc_1.50.0          
  [9] vctrs_0.6.5               memoise_2.0.1            
 [11] DelayedMatrixStats_1.26.0 tinytex_0.51             
 [13] htmltools_0.5.8.1         S4Arrays_1.4.1           
 [15] curl_5.2.1                BiocNeighbors_1.22.0     
 [17] SparseArray_1.4.8         sass_0.4.9               
 [19] bslib_0.7.0               cachem_1.1.0             
 [21] igraph_2.0.3              lifecycle_1.0.4          
 [23] ggside_0.3.1              pkgconfig_2.0.3          
 [25] rsvd_1.0.5                Matrix_1.7-0             
 [27] R6_2.5.1                  fastmap_1.2.0            
 [29] GenomeInfoDbData_1.2.12   digest_0.6.35            
 [31] colorspace_2.1-0          AnnotationDbi_1.66.0     
 [33] dqrng_0.4.1               irlba_2.3.5.1            
 [35] RSQLite_2.3.7             beachmat_2.20.0          
 [37] filelock_1.0.3            labeling_0.4.3           
 [39] fansi_1.0.6               httr_1.4.7               
 [41] abind_1.4-5               compiler_4.4.0           
 [43] bit64_4.0.5               withr_3.0.0              
 [45] BiocParallel_1.38.0       viridis_0.6.5            
 [47] DBI_1.2.2                 highr_0.11               
 [49] rappdirs_0.3.3            DelayedArray_0.30.1      
 [51] rjson_0.2.21              bluster_1.14.0           
 [53] tools_4.4.0               vipor_0.4.7              
 [55] beeswarm_0.4.0            glue_1.7.0               
 [57] grid_4.4.0                cluster_2.1.2            
 [59] generics_0.1.3            gtable_0.3.5             
 [61] BiocSingular_1.20.0       ScaledMatrix_1.12.0      
 [63] metapod_1.12.0            xml2_1.3.6               
 [65] utf8_1.2.4                XVector_0.44.0           
 [67] ggrepel_0.9.5             BiocVersion_3.19.1       
 [69] pillar_1.9.0              stringr_1.5.1            
 [71] limma_3.60.2              dplyr_1.1.4              
 [73] lattice_0.20-45           bit_4.0.5                
 [75] tidyselect_1.2.1          locfit_1.5-9.9           
 [77] Biostrings_2.72.0         gridExtra_2.3            
 [79] bookdown_0.39             edgeR_4.2.0              
 [81] svglite_2.1.3             xfun_0.44                
 [83] statmod_1.5.0             stringi_1.8.4            
 [85] UCSC.utils_1.0.0          yaml_2.3.8               
 [87] evaluate_0.23             codetools_0.2-18         
 [89] tibble_3.2.1              BiocManager_1.30.23      
 [91] cli_3.6.2                 systemfonts_1.1.0        
 [93] munsell_0.5.1             jquerylib_0.1.4          
 [95] Rcpp_1.0.12               png_0.1-8                
 [97] parallel_4.4.0            blob_1.2.4               
 [99] sparseMatrixStats_1.16.0  viridisLite_0.4.2        
[101] scales_1.3.0              crayon_1.5.2             
[103] rlang_1.1.3               KEGGREST_1.44.0          </code></pre>
</div>
<div id="references" class="section level1" number="8">
<h1><span class="header-section-number">8</span> References</h1>
</div>



</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": {
      styles: {
        ".MathJax_Display": {
           "text-align": "center",
           padding: "0px 150px 0px 65px",
           margin: "0px 0px 0.5em"
        },
        "@media screen and (max-width: 991px)": {
            ".MathJax_Display": {
               "text-align": "center",
               padding: "0 0 0 0"
            }
         }
      }
    }
  });
</script>
<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

<script type="text/javascript">
$(document).ready(function ()  {
  
  // Map "enter" keypress to the same action as a cursor click
  function navigateLink(e) {
    if (e.key === "Enter") {
      $(this).trigger("click");
    }
  }

  var toc_items = document.querySelectorAll(".tocify-item");
  for (var i = 0; i < toc_items.length; i++) {
    // The link role tells screen readers this is for navigation
    toc_items.item(i).setAttribute("role", "link");
    // tabindex = 0 allows selection via keyboard tab presses
    toc_items.item(i).setAttribute("tabindex", "0");
    // Listen for "Enter" keypress when item is selected
    toc_items.item(i).addEventListener("keydown", navigateLink);
  }
});
</script>

</body>
</html>
